### ISSUE TITLE
Generate expressions for findings instances from Web IDL

### ISSUE THREAD

**Issue by foolip** [foolip/mdn-bcd-collector#517]
Just by looking at the sum of all IDL we can know that `navigator.mediaDevices` is an instance of `MediaDevices`. Automate the generation of all such expressions to get more accurate test results than by looking at just the prototypes.

**Comment by foolip** [comment]
I gave it a quick try in https://github.com/foolip/mdn-bcd-collector/pull/518 to see how many instances this would get us. Turns out, not as many as I'd hoped:

self (AnimationWorkletGlobalScope)
self (LayoutWorkletGlobalScope)
self (PaintWorkletGlobalScope)
self (Window)
self.document (Document)
self.document.implementation (DOMImplementation)
self.document.namedFlows (NamedFlowMap)
self.document.images (HTMLCollection)
self.document.embeds (HTMLCollection)
self.document.plugins (HTMLCollection)
self.document.links (HTMLCollection)
self.document.forms (HTMLCollection)
self.document.scripts (HTMLCollection)
self.document.anchors (HTMLCollection)
self.document.applets (HTMLCollection)
self.document.all (HTMLAllCollection)
self.document.permissionsPolicy (PermissionsPolicy)
self.document.timeline (DocumentTimeline)
self.document.domConfig (DOMConfiguration)
self.document.domConfig.parameterNames (DOMStringList)
self.document.featurePolicy (FeaturePolicy)
self.document.fonts (FontFaceSet)
self.document.styleSheets (StyleSheetList)
self.document.children (HTMLCollection)
self.location (Location)
self.location.ancestorOrigins (DOMStringList)
self.location.fragmentDirective (FragmentDirective)
self.history (History)
self.customElements (CustomElementRegistry)
self.locationbar (BarProp)
self.menubar (BarProp)
self.personalbar (BarProp)
self.scrollbars (BarProp)
self.statusbar (BarProp)
self.toolbar (BarProp)
self.navigator (Navigator)
self.navigator.clipboard (Clipboard)
self.navigator.contacts (ContactsManager)
self.navigator.credentials (CredentialsContainer)
self.navigator.geolocation (Geolocation)
self.navigator.gpu (GPU)
self.navigator.scheduling (Scheduling)
self.navigator.keyboard (Keyboard)
self.navigator.mediaCapabilities (MediaCapabilities)
self.navigator.mediaDevices (MediaDevices)
self.navigator.mediaSession (MediaSession)
self.navigator.permissions (Permissions)
self.navigator.presentation (Presentation)
self.navigator.serial (Serial)
self.navigator.serviceWorker (ServiceWorkerContainer)
self.navigator.wakeLock (WakeLock)
self.navigator.bluetooth (Bluetooth)
self.navigator.hid (HID)
self.navigator.usb (USB)
self.navigator.xr (XRSystem)
self.navigator.plugins (PluginArray)
self.navigator.mimeTypes (MimeTypeArray)
self.navigator.fonts (FontManager)
self.navigator.connection (NetworkInformation)
self.navigator.storage (StorageManager)
self.navigator.userAgentData (NavigatorUAData)
self.navigator.locks (LockManager)
self.applicationCache (ApplicationCache)
self.cookieStore (CookieStore)
self.screen (Screen)
self.screen.orientation (ScreenOrientation)
self.external (External)
self.speechSynthesis (SpeechSynthesis)
self.trustedTypes (TrustedTypePolicyFactory)
self.trustedTypes.emptyHTML (TrustedHTML)
self.trustedTypes.emptyScript (TrustedScript)
self.visualViewport (VisualViewport)
self.styleMedia (StyleMedia)
self.clientInformation (Navigator)
self.clientInformation.clipboard (Clipboard)
self.clientInformation.contacts (ContactsManager)
self.clientInformation.credentials (CredentialsContainer)
self.clientInformation.geolocation (Geolocation)
self.clientInformation.gpu (GPU)
self.clientInformation.scheduling (Scheduling)
self.clientInformation.keyboard (Keyboard)
self.clientInformation.mediaCapabilities (MediaCapabilities)
self.clientInformation.mediaDevices (MediaDevices)
self.clientInformation.mediaSession (MediaSession)
self.clientInformation.permissions (Permissions)
self.clientInformation.presentation (Presentation)
self.clientInformation.serial (Serial)
self.clientInformation.serviceWorker (ServiceWorkerContainer)
self.clientInformation.wakeLock (WakeLock)
self.clientInformation.bluetooth (Bluetooth)
self.clientInformation.hid (HID)
self.clientInformation.usb (USB)
self.clientInformation.xr (XRSystem)
self.clientInformation.plugins (PluginArray)
self.clientInformation.mimeTypes (MimeTypeArray)
self.clientInformation.fonts (FontManager)
self.clientInformation.connection (NetworkInformation)
self.clientInformation.storage (StorageManager)
self.clientInformation.userAgentData (NavigatorUAData)
self.clientInformation.locks (LockManager)
self.indexedDB (IDBFactory)
self.crypto (Crypto)
self.crypto.subtle (SubtleCrypto)
self.performance (Performance)
self.performance.eventCounts (EventCounts)
self.performance.timing (PerformanceTiming)
self.performance.navigation (PerformanceNavigation)
self.performance.memory (MemoryInfo)
self.caches (CacheStorage)
self.sessionStorage (Storage)
self.localStorage (Storage)
self (DedicatedWorkerGlobalScope)
self (SharedWorkerGlobalScope)
self (ServiceWorkerGlobalScope)
self.clients (Clients)
self.registration (ServiceWorkerRegistration)
self.registration.navigationPreload (NavigationPreloadManager)
self.registration.backgroundFetch (BackgroundFetchManager)
self.registration.sync (SyncManager)
self.registration.index (ContentIndex)
self.registration.cookies (CookieStoreManager)
self.registration.paymentManager (PaymentManager)
self.registration.paymentManager.instruments (PaymentInstruments)
self.registration.periodicSync (PeriodicSyncManager)
self.registration.pushManager (PushManager)
self.serviceWorker (ServiceWorker)
self.cookieStore (CookieStore)
self (AudioWorkletGlobalScope)
self (RTCIdentityProviderGlobalScope)
self.rtcIdentityProvider (RTCIdentityProviderRegistrar)

There's duplication in that list, with lots of ways to find a `BarProp` or `HTMLCollection` and `clientInformation` is the same object as `navigator`, which this script can't see.

There are many more interfaces that can't be created without custom code, so I wonder if we should just add these as custom tests and not both with the automatic generation part. @vinyldarkscratch WDYT?
